sdks/python/apache_beam/internal/apiclient.py

```diff
@@ -374,18 +375,44 @@ def stage_file(self, gcs_or_local_path, file_name, stream,
 
   # TODO(silviuc): Refactor so that retry logic can be applied.
   @retry.no_retries  # Using no_retries marks this as an integration point.
-  def create_job(self, job):
-    """Submits for remote execution a job described by the workflow proto."""
-    # Stage job resources and add an environment proto with their paths.
+  def create_job(self, job, pipeline_options):
+    """Creates a job description.
+    Additionally, it may stage it and/or submit it for remote execution."""
+    self.create_job_description(job)
+
+    # Stage and submit the job when necessary
+    dataflow_job_file = pipeline_options.view_as(DebugOptions).dataflow_job_file
+    template_location = (
+        pipeline_options.view_as(GoogleCloudOptions).template_location)
+    if template_location:
+      if dataflow_job_file:
+        raise RuntimeError(
+            '--dataflow_job_file and --template_location '
+            'are mutually exclusive.')
+      gcs_or_local_path = dirname(template_location)
+      file_name = basename(template_location)
+      self.stage_file(gcs_or_local_path, file_name, StringIO(str(job)))
```

Comment:
Add a json() method to Job and call it here. (Also make __str__ delegate to this method.)